### PR TITLE
Improve in out unit tests

### LINKED ISSUE #848: 📋 [TASK] - Improve Tests in In/Out Unit Tests
### Parent Feature

#844

### Task Description

As an AaC Test Developer, I want to remove nose2 usage from my unit tests, so that I can more easily use pytest and VSCode for debugging unit tests.

### Acceptance Criteria

- [x] Remove nose2 package from unit tests.
- [x] Replace with equivalent pytest package.
- [x] Update test logic as necessary to expand test coverage. 
  - [x] Happy path and sad path at a minimum
- [ ] (Stretch) Incorporate a behave feature file for testing if applicable.

### Unresolved questions

_No response_

### PR DESCRIPTION
**Pull request by jasoncbryant**
# Description
Used parameterized to create param sets for test_path.py nominal (success) and off-nominal (fail) tests.

# Linked Items:

Closes #848 

### Added

- Added new dependency to parameterized to pyproject.toml
- Added an initial set of off-nominal (fail) tests to test_paths.py

### Changed

- N/A

### Deprecated

- N/A

### Removed

- Removed remaining nose2 from AaC, particularly in_out tests.
- Removed existing temporary nose2 workaround.

### Fixed

- N/A

### Security

- N/A

# Checklist:

- [x] I updated project documentation to reflect my changes.
- [x] My changes generate no new warnings.
- [x] I updated new and existing unit tests to account for my changes.
- [x] I linked the associated item(s) to be closed.
- [x] I bumped the version.
- [x] I added the labels corresponding to my changes.

### CODE REVIEW

File: python/tests/test_aac/in_out/test_paths.py [hunk 1]
```diff
@@ -19,46 +17,36 @@ def _get_working_directory_parent_directory(level_of_directories_up: int = 1) ->
...
+        [f"{WORKING_TEST_DIR}/%2e%2e//src/aac/test.py",             f"{_get_working_directory_parent_directory(1)}/src/aac/test.py"],
+        [f"{WORKING_TEST_DIR}/..%2f/src/aac/test.py",               f"{_get_working_directory_parent_directory(1)}/src/aac/test.py"],
+        [f"{WORKING_TEST_DIR}/%2e%2e%5c/src/aac/test.py",           f"{_get_working_directory_parent_directory(1)}/src/aac/test.py"],
+        [f"{WORKING_TEST_DIR}/%2e%2e\/src/aac/test.py",             f"{_get_working_directory_parent_directory(1)}/src/aac/test.py"],
+        [f"{WORKING_TEST_DIR}/..%5c\/src/aac/test.py",              f"{_get_working_directory_parent_directory(1)}/src/aac/test.py"],
+        [f"{WORKING_TEST_DIR}/%252e%252e%255c\/src/aac/test.py",    f"{_get_working_directory_parent_directory(1)}/src/aac/test.py"],
+        [f"{WORKING_TEST_DIR}/..%255c\/src/aac/test.py",            f"{_get_working_directory_parent_directory(1)}/src/aac/test.py"],
+    ])
+    def test_sequence_success(self, test_path, expected_result):
+        self.assertEqual(sanitize_filesystem_path(test_path),expected_result)
```
Comment (lizzcondrey) [inline python/tests/test_aac/in_out/test_paths.py:39]:
```suggestion
        self.assertEqual(sanitize_filesystem_path(test_path), expected_result)
```

File: python/tests/test_aac/in_out/test_paths.py [hunk 2]
```diff
@@ -19,46 +17,36 @@ def _get_working_directory_parent_directory(level_of_directories_up: int = 1) ->
 
     return path_to_return
 
-
 class TestPaths(TestCase):
+    #Success test block happy path cases
```
Comment (lizzcondrey) [inline python/tests/test_aac/in_out/test_paths.py:21]:
```suggestion
    # Success test block happy path cases
```

File: python/tests/test_aac/in_out/test_paths.py [hunk 3]
```diff
@@ -19,46 +17,36 @@ def _get_working_directory_parent_directory(level_of_directories_up: int = 1) ->
...
+    @parameterized.expand([
+        [f"{WORKING_TEST_DIR}/src/aac/AAA.py",                      f"{WORKING_TEST_DIR}/src/aac/BBB.py"], #Example of input for a fail-test
+        [f"{WORKING_TEST_DIR}/src/aac/AAA.py",                      f""], #Empty directory
+        [f"{WORKING_TEST_DIR}/src/aac/AAA.py",                      f"{WORKING_TEST_DIR}/src/aac//AAA.py"], #Double forward slash test
+        [f" ",                                                      f""], #Space vs empty/null directory
+        [f" ",                                                      f" "], #Tab vs space directory
+        [f"{WORKING_TEST_DIR}/src/aac/AAA.py",                      f"{WORKING_TEST_DIR}/src/aac/AAA.py "], #Trailing space test
+        ])
+    def test_sequence_fail(self, test_path, expected_result):
+        self.assertNotEqual(sanitize_filesystem_path(test_path),expected_result) #Example of a fail-test assertNotEqual rather than assertEqual
```
Comment (lizzcondrey) [inline python/tests/test_aac/in_out/test_paths.py:51]:
```suggestion
    # Expected fail test block for sad path cases
    @parameterized.expand([
        [f"{WORKING_TEST_DIR}/src/aac/AAA.py",                      f"{WORKING_TEST_DIR}/src/aac/BBB.py"],  # Example of input for a fail-test
        [f"{WORKING_TEST_DIR}/src/aac/AAA.py",                      f""], #Empty directory
        [f"{WORKING_TEST_DIR}/src/aac/AAA.py",                      f"{WORKING_TEST_DIR}/src/aac//AAA.py"],  # Double forward slash test
        [f" ",                                                      f""],  # Space vs empty/null directory
        [f" ",                                                      f" "],  # Tab vs space directory
        [f"{WORKING_TEST_DIR}/src/aac/AAA.py",                      f"{WORKING_TEST_DIR}/src/aac/AAA.py "],  # Trailing space test
        ])
    def test_sequence_fail(self, test_path, expected_result):
        self.assertNotEqual(sanitize_filesystem_path(test_path), expected_result)
```

A handful of formatting cleanup.

File: python/tests/test_aac/in_out/test_paths.py [hunk 4]
```diff
@@ -1,12 +1,10 @@
 import os
-# from nose2.tools import params
+from parameterized import parameterized
 from unittest import TestCase
-
 from aac.in_out.paths import sanitize_filesystem_path
```
Comment (lizzcondrey) [inline python/tests/test_aac/in_out/test_paths.py:4]: I wonder if this is somewhere within the logic we should look at bubbling some file handle errors up from if it is a true file location issue.

Thoughts?

**Review by lizzcondrey — CHANGES_REQUESTED** [review]
Just some format clean-up for the comments.

Because this is within a unit test the linting ignores comment requirements. But generally, there are two spaces before and one space after the `#` for inline comments.

**Comment by lizzcondrey** [comment]
Don't forget to capture any odd or missing behaviors you found during this work. Not seen notices for any issues being made and you had commented that there might be some missing cases.

**Review by lizzcondrey — CHANGES_REQUESTED** [review]
Missing spaces between arguments in the assertion checks. 

But all tests pass for unittest, tox, and pytest. Good job getting this expanded and nose2 dropped.

File: python/tests/test_aac/in_out/test_paths.py (hunk 4 above)
Comment (jasoncbryant) [reply python/tests/test_aac/in_out/test_paths.py:4]: That looks like a good place.

**Review by lizzcondrey — APPROVED** [review]

**Pull request merged**